### ISSUE TITLE
Downloading the CsvHelper package result in a conflict with Microsoft.CSharp.dll when targeting NetStandard 2.0

### ISSUE THREAD

**Issue by Mythique** [GlitchEnzo/NuGetForUnity#343]
## Description

Downloading the CsvHelper package will download Microsoft.CSharp.4.5.0, which result in a conflict :

> Microsoft (R) Visual C# Compiler version 2.9.1.65535 (9d34608e)
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
> error CS1703: Multiple assemblies with equivalent identity have been imported: 'C:\Users\Developpeur\Documents\Source\Projets\Socotec2_Unity\Assets\Packages\Microsoft.CSharp.4.5.0\lib\netstandard2.0\Microsoft.CSharp.dll' and 'C:\Program Files\Unity\Hub\Editor\2019.4.10f1\Editor\Data\MonoBleedingEdge\lib\mono\4.7.1-api\Microsoft.CSharp.dll'. Remove one of the duplicate references.

- **NuGet Package:** CsvHelper 15.0.5
- **NuGetForUnity Version:** 2.0.1
- **Unity Version:** 2019.4.10f1
- **Operating System:** Windows 10

**Comment by jwittner** [comment]
This is a common problem when targetting netstandard 2.0 right now as Unity includes some libraries by default that are normally installed via NuGet. If you can target .Net 4.x that has resolved this issue in the past. If you'd like to contribute a solution, I'm hoping for a soluation that scans the current Unity editor to determine libraries already imported and then delete any conflicting libraries installed by NuGetForUnity.

**Comment by Mythique** [comment]
Thanks for the input, I will try switching to .net 4.x

In the MSBuildForUnity tool they have [an exclusion list](https://github.com/microsoft/MSBuildForUnity/blob/master/Source/MSBuildTools.Unity.NuGet/Publish.props).

**Comment by jwittner** [comment]
I've seen that exclusion list, my issue is that it doesn't handle different versions of Unity well and creates a management burden every time a new Unity version comes out with new .Net API support.

**Comment by hawkerm** [comment]
I think I've hit a similar issue trying to include the `Microsoft.Toolkit.Mvvm` package:

```
Assembly 'Assets/Packages/Microsoft.Bcl.AsyncInterfaces.1.1.1/lib/netstandard2.0/Microsoft.Bcl.AsyncInterfaces.dll' will not be loaded due to errors:
Microsoft.Bcl.AsyncInterfaces references strong named System.Threading.Tasks.Extensions in a different folder, versions has to match. Assembly references: 4.2.0.0 Found in project: 4.2.0.1.

Assembly 'Assets/Packages/System.Memory.4.5.4/lib/netstandard2.0/System.Memory.dll' will not be loaded due to errors:
System.Memory references strong named System.Buffers in a different folder, versions has to match. Assembly references: 4.0.2.0 Found in project: 4.0.3.0.

Assembly 'Assets/Packages/Microsoft.Toolkit.Mvvm.7.0.0-preview4/lib/netstandard2.0/Microsoft.Toolkit.Mvvm.dll' will not be loaded due to errors:
Microsoft.Toolkit.Mvvm references strong named System.ComponentModel.Annotations in a different folder, versions has to match. Assembly references: 4.2.0.0 Found in project: 4.2.1.0.
Microsoft.Toolkit.Mvvm references strong named System.Buffers in a different folder, versions has to match. Assembly references: 4.0.2.0 Found in project: 4.0.3.0.
```

I tried to add an `app.config` file to do a [binding redirect](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/redirect-assembly-versions), but not sure if it's being picked up (not sure where to put it) or if I don't have it setup correctly.

This package only supports .NET Standard 2.0 as a target, so does this mean I'm stuck? Or can I create a similar exclusion list in my Unity project somehow as well?

**Comment by hawkerm** [comment]
I was able to work around the conflicts by copying all the `netstandard2.0` dlls to the `plugins` folder of my project instead.

Is there an option to get NuGetForUnity to just drop dlls flat to the `plugins` folder instead of keeping the whole NuGet package in the `Packages` folder? It'd be great to keep the ease-of-use and ability to update via the UI.

**Comment by jwittner** [comment]
Right now there's not currently an exclusions list supported in NuGetForUnity - we're open to PRs though, and I can provide guidance on how one might implement that if you're interested in making one. =)

Packages that only target .Net Standard 2.0 will still work if you configure Unity to target .Net 4.x. While NuGetForUnity will prefer .Net Framework lib folders if available, the .Net Standard ones will be used if no Framework ones area available.

**Comment by marcus-robinson** [comment]
@jwittner I'm experiencing a similar issue due to the library I'm importing having both x86 and 64bit DLL's (which Unity does not like - "Multiple plugins with the same name" error). What is the suggested workaround for this? Do devs need to manually delete one of the DLLs after N4U downloads them?

**Comment by jwittner** [comment]
Right now the current work around is to use .Net 4.x instead of .Net Standard 2.0. Help testing #357, which seeks to resolve this issue at it's core, would go a long way to getting this fixed for everyone. =)

**Comment by jwittner** [comment]
Note: may be resolved by #357

**Comment by KonH** [comment]
Looks like it will be resolved by #357 (tested on Unity 2019.4.0f1, .NET Standard 2.0 subset, MacOS)

**Comment by OmarWKH** [comment]
Faced this in Unity 2022.3.3f1 with NuGetForUnity 3.1.0 installing CsvHelper 30.0.1.
Api compatibility level is .NET Standard 2.1.

**Comment by JoC0de** [comment]
I just released version 3.1.3 which also contains some fixes that probably resolve this issue.
